docs/content/config/advanced/auth-lua.md

```diff
@@ -0,0 +1,194 @@
+  if auth_response:status() == http_status_ok
+  then
+    returnStatus = dovecot.auth.PASSDB_RESULT_OK
+    returnDesc = "nopassword=y"
+  elseif auth_response:status() == http_status_failure
+  then
+    returnStatus = dovecot.auth.PASSDB_RESULT_PASSWORD_MISMATCH
+    returnDesc = ""
+  end
+  return returnStatus, returnDesc
+```
+
+Replace the hostname in the URL to the actual hostname of Nextcloud.
+
+Dovecot [provides an HTTP client for use in Lua](https://doc.dovecot.org/admin_manual/lua/#dovecot.http.client). Aside of that, Lua by itself is pretty barebones. It chooses library compactness over included functionality. You can see that in the inefficiently typed regular expression in `is_nextcloud_apppassword()`, which is used because [Lua does not offer full support for regular expressions](https://www.lua.org/pil/20.2.html). You can also see Lua's limited functionality in that a separate library is referenced to add support for Base64 encoding, which is required for [HTTP basic access authentication](https://en.wikipedia.org/wiki/Basic_access_authentication). This library (also a Lua script) is not included. It must be downloaded and stored in the same directory:
+
+```bash
+cd ./docker-data/dms/config/dovecot/lua
+curl -JLO https://raw.githubusercontent.com/iskolbin/lbase64/master/base64.lua
+```
+
+Only use native (pure Lua) libraries as dependencies, such as `base64.lua` from the example. This ensures maximum compatibility. Performance is less of an issue since Lua scripts written for Dovecot probably won't be long or complex, and there won't be a lot of data processing by Lua itself. To see which Lua version is used by Dovecot if you plan to do something that is version dependent, run:
+
+```bash
+docker exec mailserver strings /usr/lib/dovecot/libdovecot-lua.so|grep '^LUA_'
+```
+
+## Debugging a Lua script
+
+Aside of succeeded and failed authentication attempts for any passdb backend, Dovecot also logs Lua scripting errors and messages send to Dovecot's [Lua API log functions](https://doc.dovecot.org/admin_manual/lua/#dovecot.i_debug). The combined DMS log (including that of Dovecot) can be viewed using `docker logs mailserver` (*where `mailserver` is the container name*). If the log is too noisy (due to other processes in the container also logging to it), `docker exec mailserver cat /var/log/mail/mail.log` can be used to view the log of Dovecot and Postfix specifically. If working with HTTP in Lua, setting `debug = true;` when initiating `dovecot.http.client` will create debug log messages for every HTTP request and response.
+
+Note that Lua runs compiled bytecode, and that scripts will be compiled when they are initially started. Once compiled, the bytecode is cached and changes in the script will not be processed. [Restart sub-service Dovecot][docs-faqalterdms] using `docker exec mailserver supervisorctl restart dovecot` to have Dovecot load a changed Lua script.
```

Comment:
Could you confirm that restarting the Dovecot service completely is required? Or is it sufficient to run `dovecot reload`? (_which is what [`check-for-changes.sh` does](https://github.com/docker-mailserver/docker-mailserver/blob/82c38f242648290ab109de8e3994ea7564120deb/target/scripts/check-for-changes.sh#L57)_)